E2E/tests/voting.test.ts

```diff
@@ -0,0 +1,189 @@
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user2);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
+    game.vote(chessPiece.knight, chessTile.e4);
+    game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  });
+  I.waitForElement(game.locator.playersList.playerDoneIcon(username.user2));
+  game.voteIsNotVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+});
+
+Scenario('Two users with same name vote as separate players', async ({ I, login, username, roles, game, chessPiece, chessTile }) => {
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user1);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
+    game.vote(chessPiece.knight, chessTile.e4);
+    game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  });
+  I.waitForElement(game.locator.playersList.playerDoneIcon(username.user1));
+  game.voteIsNotVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  game.vote(chessPiece.queen, chessTile.f5);
+  I.waitForText('Game complete');
+  game.voteIsVisible(chessPiece.queen, chessTile.f5, chessPiece.queenValue);
+  game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  I.see(username.user1, game.locator.playersList.firstUserInList);
+  I.see(username.user1, game.locator.playersList.secondUserInList);
+  session(username.user2, () => {
+    game.voteIsVisible(chessPiece.queen, chessTile.f5, chessPiece.queenValue);
+    game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+    I.see(username.user1, game.locator.playersList.firstUserInList);
+    I.see(username.user1, game.locator.playersList.secondUserInList);
+  });
+});
+
+Scenario('3 players place different figures on different board squares', async ({ I, login, username, roles, game, chessPiece, chessTile }) => {
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user2);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
+  });
+  session(username.user3, () => {
+    login.loginIntoCreatedGameRoom(url, username.user3);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
+  });
+  game.vote(chessPiece.king, chessTile.a1);
+  game.voteIsVisible(chessPiece.king, chessTile.a1, chessPiece.kingValue);
+  session(username.user2, () => {
+    game.voteIsNotVisible(chessPiece.king, chessTile.a1, chessPiece.kingValue);
+    game.vote(chessPiece.bishop, chessTile.b3);
+    game.voteIsVisible(chessPiece.bishop, chessTile.b3, chessPiece.bishopValue);
+  });
+  session(username.user3, () => {
```

Comment:
I'm brainstorming how we could make tests a bit more readable.

What do you think if we were to introduce a class for `User` to make tests easier to reason about.
Maybe something like this could do the trick?
```
class User {
  name: string;

  constructor(name) {
    this.name = name;
  }

  login(chessPiece, tile) {
    session(this.name, () => {
      <...>
    })
  }

  vote(chessPiece, tile) {
    session(this.name, () => {
      <...>
    })
  }
  
  canSee(chessPiece, tile) {
    session(this.name, () => {
      <...>
    })
  }
}


...


const user1 = new User('user1');
user1.login();
user1.vote(knight, a1);
user1.canSee(knight, a1);
```

Reply:
We could apply the same logic for the gameboard as well.